src/Fmt_ast.ml

```diff
@@ -2745,14 +2743,16 @@ and fmt_type_declaration c ?(pre = "") ?(suf = ("" : _ format)) ?(brk = suf)
                     $ fmt_core_type c (sub_typ ~ctx t2) )) )) )
   in
   let doc, atrs = doc_atrs ptype_attributes in
+  let maybe_box = if List.is_empty ptype_params then Fn.id else hvbox 0 in
   Cmts.fmt c loc @@ Cmts.fmt c ptype_loc
   @@ hvbox 0
        ( fmt_docstring c ~epi:(fmt "@\n") doc
        $ hvbox 0
            ( hvbox 2
-               ( str pre
-               $ fmt_tydcl_params c ctx ptype_params
-               $ (match fmt_name with Some pp -> pp | None -> str txt)
+               ( open_hvbox 2 $ str pre
```

Comment:
This open should pair with the closes in `fmt_manifest_kind`? I don't like this.
I suggest you pass this formatting to `fmt_manifest_kind` and let it handle boxes, without explicit open/close.

Reply:
done